src/libNetwork/Guard.cpp

```diff
@@ -221,6 +222,28 @@ void Guard::AddToExclusionList(const uint128_t& ft, const uint128_t& sd) {
   }
 }
 
+void Guard::AddDSGuardToBlacklistExcludeList(const DequeOfNode& dsComm) {
+  if (GUARD_MODE) {
+    unsigned int dsIndex = 0;
+    for (const auto& i : dsComm) {
+      if (dsIndex < GetNumOfDSGuard()) {
```

Comment:
may be we don't need check with `dsIndex`. we already check it below.

Reply:
the else should be in the outer `if`. Will make the change

Reply:
I guess it just let it break out of the loop earlier